File: 3619-adjacent-increasing-subarrays-detection-ii/adjacent-increasing-subarrays-detection-ii.cpp
class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        vector<int> store;
        int cnt=1;
        int n=nums.size();
        for(int i=0; i<n-1; i++){
            if(nums[i] < nums[i+1]){
                cnt++;
            }
            else{
                store.push_back(cnt);
                cnt=1;
            }
        }
        store.push_back(cnt);
        int maxCon = INT_MIN;
        int minAm = INT_MAX;
        for(int i=0; i<store.size()-1; i++){
            minAm = min(store[i], store[i+1]);
            maxCon=max(maxCon, minAm);
        }
        for(auto it: store){
            int value = it % 2;
            it -= value;
            maxCon = max(maxCon, it / 2);
        }
        return maxCon == INT_MIN ? 1 : maxCon;
    }
};

File: 3619-adjacent-increasing-subarrays-detection-ii/adjacent-increasing-subarrays-detection-ii.py
class Solution:
    def maxIncreasingSubarrays(self, nums: List[int]) -> int:
        n = len(nums)
        store = []
        cnt = 1
        for i in range(n-1):
            if nums[i] < nums[i+1]:
                cnt+=1
            else:
                store.append(cnt)
                cnt = 1
        store.append(cnt)
        minAm = float('inf')
        print(store)
        maxCon = float('-inf')
        for i in range(len(store)-1):
            minAm = min(store[i], store[i+1])
            maxCon = max(maxCon, minAm)
        for i in range(len(store)):
            value = store[i] % 2
            store[i] -= value
            maxCon = max(maxCon, store[i]//2)
        return maxCon if maxCon != float('-inf') else 1
